### PR TITLE
Make x-ray size adapt to DB query speed

### PR DESCRIPTION
**Pull request by sbelak**
This makes x-rays (somewhat) respect DB performance. If either `auto_run_queries` is disabled, or we are consistently seeing slow running queries (thershold set at >5s for the 90th percentile), show only either the "summary" section of an x-ray (these tend do be just a few cards and highly aggregated), or the first 3 cards if the x-ray does not have a "summary" section.

### CODE REVIEW

**Review by camsaul — APPROVED** [review]

**Review by camsaul — APPROVED** [review]
WFM

File: src/metabase/automagic_dashboards/adaptive_dashboard_size.clj [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  (:require [bigml.histogram.core :as hist]
+            [clojure.core.memoize :as memo]
+            [metabase.models
+             [database :refer [Database]]
+             [query-execution :refer [QueryExecution]]]
+            [metabase.sync.analyze.fingerprint.fingerprinters :refer [histogram]]
+            [redux.core :as redux]
+            [toucan.db :as db]))
+
+(def ^:private ^Long long-running-90th-percentile-threshold
```
Comment (salsakran) [inline src/metabase/automagic_dashboards/adaptive_dashboard_size.clj]: As mentioned off-line, I don't like the idea of x-rays having their own sense of whether a given card/database is slow. 

This should be centralized somewhere other than the automagic dashboard code.

File: src/metabase/automagic_dashboards/adaptive_dashboard_size.clj [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                        (redux/post-complete
+                         histogram
+                         (fn [h]
+                           ((hist/percentiles h 0.9) 0.9)))
+                        (db/select-field :running_time QueryExecution :database_id %))
+            :ttl/threshold running-time-cache-ttl))
+
+(defn- slow-db?
+  [db-id]
+  (some-> db-id running-time-90th-percentile (> long-running-90th-percentile-threshold)))
```
Comment (salsakran) [inline src/metabase/automagic_dashboards/adaptive_dashboard_size.clj]: we should use the same db setting that determines whether or not to automatically run queries in view mode, not replicate it in a subtly different way here.

**Review by salsakran — CHANGES_REQUESTED** [review]

File: src/metabase/automagic_dashboards/adaptive_dashboard_size.clj (hunk 2 above)
Comment (sbelak) [reply src/metabase/automagic_dashboards/adaptive_dashboard_size.clj]: It uses both, the flag (line 41) and a heuristic

**Comment by sbelak** [comment]
Need to factor out the performance metrics.

**Comment by rlotun** [comment]
closing due to age.

**Pull request closed without merging**